mwdb/web/src/components/ShowObject/Views/RelatedFilesTab.js

```diff
@@ -0,0 +1,204 @@
+import { humanFileSize } from "@mwdb-web/commons/helpers";
+import ReactModal from "react-modal";
+
+async function updateRelatedFiles(api, context) {
+    const { setObjectError, updateObjectData } = context;
+    try {
+        let response = await api.getListOfRelatedFiles(context.object.sha256);
+        updateObjectData({
+            related_files: response.data.related_files,
+        });
+    } catch (error) {
+        setObjectError(error);
+    }
+}
+
+function ShowRelatedFiles() {
+    const api = useContext(APIContext);
+    const context = useContext(ObjectContext);
+    const { setObjectError, updateObjectData } = context;
+
+    function RelatedFileItem({ file_name, file_size, sha256 }) {
+        return (
+            <tr>
+                <td>{file_name}</td>
+                <td>{humanFileSize(file_size)}</td>
+                <td>
+                    <Link
+                        to={`/file/${context.object.sha256}/related_files`}
+                        className="nav-link"
+                        onClick={async () => {
+                            let content = await api.downloadRelatedFile(sha256);
+                            let blob = new Blob([content.data], {
+                                type: "application/octet-stream",
+                            });
+                            let tempLink = document.createElement("a");
+                            tempLink.style.display = "none";
+                            tempLink.href = window.URL.createObjectURL(blob);
+                            tempLink.download = file_name;
+                            tempLink.click();
+                        }}
+                    >
+                        <FontAwesomeIcon icon={faDownload} size="1x" />
+                        &nbsp;Download
+                    </Link>
+                </td>
+                <td>
+                    <Link
+                        to={`/file/${context.object.sha256}/related_files`}
+                        className="nav-link"
+                        onClick={async () => {
+                            await api.deleteRelatedFile(sha256);
+                            updateRelatedFiles(api, context);
+                        }}
+                    >
+                        <FontAwesomeIcon icon={faTrash} size="1x" />
+                        &nbsp;Remove
+                    </Link>
+                </td>
+            </tr>
+        );
+    }
+
+    const getRelatedFiles = useCallback(updateRelatedFiles, [
+        api,
+        setObjectError,
+        updateObjectData,
+        context.object.sha256,
+    ]);
+
+    // JS throws a warning "Line 90:8:  React Hook useEffect has missing dependencies: 'api' and 'context'"
+    // Those dependencies are skipped on purpose
+    // To disable this warning I used 'eslint-disable-next-line'
+    useEffect(() => {
+        getRelatedFiles(api, context);
+        // eslint-disable-next-line
+    }, [getRelatedFiles]);
```

Comment:
Actually that warning might be correct, because we're falling into trap of closure (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures).

```
> function render(x) {
  let a = x;
  return () => a;
}

> fn = render(5);
> render(10)
> fn()
5 
```

`useEffect` actually does the same thing and stores that function to be used for next renders (unless dependency will change) which may lead to problems. The problem is fixed here by correct use of `useCallback` which allows to recreate that function when api and `context.sha256` changes, but still we should write code that passes eslint check.

```javascript
    const { setObjectError, updateObjectData, object: {sha256} } = context;
    
    async function updateRelatedFiles() {
    // indentation to be fixed ofc
    try {
        let response = await api.getListOfRelatedFiles(sha256);
        updateObjectData({
            related_files: response.data.related_files,
        });
    } catch (error) {
        setObjectError(error);
    }
}

    const getRelatedFiles = useCallback(updateRelatedFiles, [
        api,
        setObjectError,
        updateObjectData,
        sha256,
    ]);

    useEffect(() => {
        getRelatedFiles();
    }, [getRelatedFiles]);
```

That one should work without problems

Reply:
I've tried this implementation while writing previous versions
The problem is that I need function updateRelatedFiles() to be visible outside of ShowRelatedFiles(), so I cannot make it a nested function
I use updateRelatedFiles() in handleSubmit(), which needs access to ReactModel, which is opened by ObjectAction located in RelatedFilesTab()